### PR TITLE
[MacOS] Pin PostgreSQL version installed in the toolset

### PR DESCRIPTION
**Pull request by shilovmaksim**
# Description
We need to move PostgreSQL along with some other tools to the toolset on all OSes to make versions control more granular.
## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/postgresql.sh [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 source ~/utils/utils.sh
 
-#Install latest version of postgresql
-brew_smart_install "postgres"
+# Fetch postgresql version to install from the toolset
+toolsetVersion=$(get_toolset_value '.postgresql.version')
 
-#Service postgresql should be started before use.
+# Search for the latest version of the postgresql corresponding to version from the toolset
+versionToInstall=$(brew search /postgresql@$toolsetVersion/ | awk -F' ' '{print $1}' | tail -1)
```
Comment (al-cheb) [inline images/macos/provision/core/postgresql.sh:8]: Why can't we use just `brew_smart_install postgresql@$versionToInstall`?

File: images/macos/tests/Databases.Tests.ps1 [hunk 2]
```diff
@@ -9,11 +9,11 @@ Describe "MongoDB" {
...
-    It "PostgreSQL-Client" {
-        "psql --version" | Should -ReturnZeroExitCode
-    }
-
-    It "PostgreSQL-Server" {
-        "pg_config --version" | Should -ReturnZeroExitCode
+    It "PostgreSQL version should correspond to the version in the toolset" {
+        $toolsetVersion = Get-ToolsetValue 'postgresql.version'
+        # Client version
+        (&psql --version).split(" ")[-1] | Should -BeLike "$toolsetVersion*"
```
Comment (al-cheb) [inline images/macos/tests/Databases.Tests.ps1:15]: nitpick: remove &

Comment (al-cheb) [reply images/macos/tests/Databases.Tests.ps1:15]: nitpick: split(" ") -> split()

File: images/macos/provision/core/postgresql.sh (hunk 1 above)
Comment (shilovmaksim) [reply images/macos/provision/core/postgresql.sh:8]: Indeed. I think it's just a precaution. But if you speak so too, we can omit it safely.

Comment (shilovmaksim) [reply images/macos/provision/core/postgresql.sh]: Updated

File: images/macos/tests/Databases.Tests.ps1 (hunk 2 above)
Comment (shilovmaksim) [reply images/macos/tests/Databases.Tests.ps1]: Done

**Review by miketimofeev — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**